frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/ApplicationMaster.java

```diff
@@ -526,14 +522,33 @@ private Boolean testContainerNode(String containerId, String containerHostName)
     return true;
   }
 
-  private Boolean testContainer(Container container) {
+  private Boolean testContainer(Container container, TaskStatus taskStatus) throws Exception {
```

Comment:
>, TaskStatus taskStatus [](start = 51, length = 23)

String taskRoleName

Reply:
no much different.

---
In reply to: [174689126](https://github.com/Microsoft/pai/pull/318#discussion_r174689126) [](ancestors = 174689126)

Reply:
pass in minimal things

---
In reply to: [175336556](https://github.com/Microsoft/pai/pull/318#discussion_r175336556) [](ancestors = 175336556,174689126)